x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -231,6 +234,15 @@ private static void checkBinaryComparison(LogicalPlan p, Failures failures) {
         });
     }
 
+    private static void checkInsist(LogicalPlan p, Failures failures) {
+        if (p instanceof Insist i) {
+            LogicalPlan child = i.child();
+            if ((child instanceof EsRelation || child instanceof Insist) == false) {
+                failures.add(fail(i, "[insist] can only be used after [from] or [insist] commands, but was [{}]", child.sourceText()));
+            }
+        }
+    }
+
```

Comment:
One more piece that could be refactored: have `Insist` implement `PostAnalysisVerificationAware` and move the check there.
Nit: I think we capitalise the commands elsewhere (including tests for this check), we should do the same here.